### DISCUSSION TITLE
Clarification on chained equations methodology

### DISCUSSION THREAD

**Discussion by VictorPorcelli** [amices/mice#556] [Missing data methodology]
Hi there, I'm trying to better understand the chained equations methodology used by mice and am particularly confused about when previously imputed values are used in subsequent imputations/iterations.

From my understanding, the process is as follows:
**Step 1:** Missing values in predictors for a variable being imputed are replaced with random draws from observed values.
**Step 2:** The variable is imputed using whatever method is specified (e.g., logistic regression).
**Step 3:** Steps 1 and 2 are repeated for each variable being imputed, except imputations from Step 2 replace random draws from Step 1. The end result is one fully imputed dataset (end of imputation 1).
**Step 4:** Steps 1-3 are repeated for a set number of imputations (defined by parameter **m**), the result of which is one iteration.
**Step 5:** Steps 1-4 are repeated for a set number of iterations (defined by parameter **maxit**).

My first question is am I understanding the process correctly?

Assuming I am, what I'm slightly unclear on but have attempted to ascertain using context clues from various readings, is whether:

(a) Subsequent imputations use imputed values rather than random draws (i.e., skip step 1) within a single iteration. For example, after Steps 1-3 are completed producing a single fully imputed dataset, and mice moves on to imputation number 2, is the process reset, or does imputation 2 use imputed values from imputation 1 in place of random draws from observed values?

(b) Subsequent iterations use imputed values rather than random draws. That is, after mice produces **m** fully imputed datasets and completes iteration 1, does imputation 1 of iteration 2 use random draws from observed values to fill in missing values in predictors, or does it use imputed values from iteration 1?

**Comment by stefvanbuuren** [comment]
Section <https://stefvanbuuren.name/fimd/sec-FCS.html> contains the pseudo code of the MICE algorithm. I leave it up you to spot the differences with your version.

**Reply by VictorPorcelli** [reply]
Thank you!
